src/core/grid.hpp

```diff
@@ -271,14 +277,9 @@ template <typename T1, typename T2> void fold_position(T1 &pos, T2 &image_box) {
  */
 inline Vector3d folded_position(Particle const &p) {
   Vector3d pos{p.r.p};
-
+  int tmp[3];
```

Comment:
You need to initialize this as it is compared with `INT_MAX`, this is also what clang-tidy does not like.